### PR TITLE
Data stream selection popover auto adjust width

### LINKED ISSUE #368: [BUG]Indices - "Data streams" drop down - name not readable when too long
**Describe the bug**

We're using data streams ([link](https://opensearch.org/docs/latest/opensearch/data-streams/)) instead of regular indices.
The name of the data streams is build by this scheme: logstream-<kubernetes_cluster>-<namespace>, which can lead to names that are too big for the drop down, see screenshot below.
There's no possibility to see the full name, except if it's selected and shown search bar left to the drop down. In case there are multiple streams that got a similar naming, the actual part that distinguishes them is truncated.

**To Reproduce**
1. Go to "Index Management" - "Indices"
2. Click on "Show data stream indices" - "Data streams"
3. Drop down is shown with the available data streams

**Expected behavior**
The drop down should expand to the full name of the data stream automatically or by increasing it manually. If done manually, this setting should be saved somehow.

**OpenSearch Version**
2.3.0

**Dashboards Version**
2.3.0

**Plugins**

image

**Screenshots**

image

**Host/Environment (please complete the following information):**
 - OS: macOS 12.6
 - Browser and version: 
 -- Google Chrome, Version 107
-- Mozilla Firefox, Version 105

### PR DESCRIPTION
**Pull request by phestecAMZN**
### Description
This fix is meant to address issue where data streams with longer names would be truncated in the filter selection popover.  The popover will now auto adjust its width based on the longest name.

### Issues Resolved
#368

### Check List
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
Screen Shot 2022-11-29 at 2 42 20 PM
Screen Shot 2022-11-29 at 2 37 25 PM

### CODE REVIEW

File: public/app.scss [hunk 1]
```diff
@@ -44,3 +47,10 @@ $euiTextColor: $euiColorDarkestShade !default;
 .state-accordion:hover {
   text-decoration: none;
 }
+
+.euiFilterGroup__popoverPanel {
```
Comment (phestecAMZN) [inline public/app.scss:51]: this block is the only change, all other diffs are already merged to main.

Comment (amsiglan) [inline public/app.scss:51]: Should we use a custom class so that only this particualr popover is styled differently?

File: public/app.scss [hunk 2]
```diff
@@ -44,3 +47,10 @@ $euiTextColor: $euiColorDarkestShade !default;
...
+  width: fit-content;
```
Comment (amsiglan) [inline public/app.scss:52]: Does max-width alone not result in the element stretching as required for the content?

File: public/app.scss (hunk 1 above)
Comment (phestecAMZN) [reply public/app.scss:51]: well, this plugin only has this specific classed popover in 2 places, and they both needed the same fix.  There is no actual code element to take a custom class name, it is auto generated from the options within the search bar filters.

Comment (phestecAMZN) [reply public/app.scss:51]: I checked every page for this class in dev tools, only comes up with the cases addressed in the PR

File: public/app.scss (hunk 2 above)
Comment (phestecAMZN) [reply public/app.scss:52]: it does not.  the Eui component has it's own defaults which seem to render that ineffective.

File: public/app.scss (hunk 1 above)
Comment (phestecAMZN) [reply public/app.scss:51]: Screen Shot 2022-11-29 at 3 24 44 PM

Comment (phestecAMZN) [reply public/app.scss:51]: getDataStreams returns an array of { value: _data stream name_}

Comment (amsiglan) [reply public/app.scss:51]: okay, let's keep it as is then

**Review by amsiglan — APPROVED** [review]

**Review by bowenlan-amzn — APPROVED** [review]

**Pull request merged**